### ISSUE TITLE
Issue: Replace Message Box with Dialog in Demo 300

### ISSUE THREAD

**Issue by francisco-milan** [abap2UI5/abap2UI5#1499]
Hi guys,

In Demo 300, which is included in this<a href="https://github.com/abap2UI5/samples/pull/402"> PR </a> , I am currently using a MessageBox to display the text **"Product was damaged along transportation."** with the following line of code:

    client->message_box_display( title = `Error description`
                                 text = `Product was damaged along transportation` ).

I would like to replace this with a Dialog to better resemble the sample provided in the UI5 Demo Kit. The intended replacement would be something like this:

    client->dialog_display( title = `Error description` 
                            text = `Product was damaged along transportation` ).

![image](https://github.com/user-attachments/assets/4fb3dcf5-faa0-4b17-96e6-26337b358ee6)

Observed Issues: There are key differences between the MessageBox and the Dialog:

![image](https://github.com/user-attachments/assets/9de6a2d2-bf59-448e-95bd-c0fcef1260de)

The MessageBox doesn't display the custom title as expected, and it shows an "information" icon instead.
The "OK" button in the MessageBox looks different compared to the one used in the UI5 Dialog.

![image](https://github.com/user-attachments/assets/d1bd07d6-94da-42c4-8d23-a814c7875bae)

Attempted Fixes: I made several adjustments to the interfaces and classes below, but the result was not as expected. Here are the lines of code that I added:

**Interface: Z2UI5_IF_CORE_TYPES**

      BEGIN OF s_dialog,
        text              TYPE string,
        title             TYPE string,
      END OF s_dialog,

**Interface: Z2UI5_IF_CLIENT**

    METHODS dialog_display
      IMPORTING
        text              TYPE clike
        title             TYPE clike OPTIONAL.

**Class: Z2UI5_CL_CORE_CLIENT**

    METHOD z2ui5_if_client~dialog_display.
      mo_action->ms_next-s_set-s_dialog = VALUE #(
                                                text              = text
                                                title             = title
                                              ).
    ENDMETHOD.

**Class: Z2UI5_CL_CORE_ACTION**

    CLEAR result->ms_next-s_set-s_dialog.

After trying to replace the MessageBox with a Dialog and clicking the "Damaged" link nothing happens, the expected behavior was not triggered.

![image](https://github.com/user-attachments/assets/20c44fa4-9f9e-4b7f-a6cf-0d48fdb03432)

Expected Outcome: The goal is to replace the MessageBox with a Dialog that:

- Works as expected when the "Damaged" link is clicked.
- Displays the custom title (without the default information icon),
- Matches the button appearance and behavior from the UI5 Dialog,

Could you please assist with guidance on implementing the Dialog correctly? I would appreciate any suggestions on what adjustments are needed to resolve the link-click issue and align the UI with the UI5 Demo Kit sample.

Thanks,
Francisco.

**Comment by choper725** [comment]
i remember at some point i implemented message box anf toast fully..

**Comment by oblomov-dev** [comment]
HI @francisco-milan,

the idea of 
client->message_toast_display(...)
client->message_box_display(...)

is to just put basic functionality in there, if possible i would prefer to avoid making changes here.

Did you already checked the popups?

for example: z2ui5_cl_pop_to_confirm
just try sample: Z2UI5_CL_DEMO_APP_151

maybe we can better adjust one of these popups or create a new popup for the dialog?

**Comment by francisco-milan** [comment]
Hi @chopper and @oblomov-dev,

I just wanted to clarify that Demo 300 is fully functional as it stands, currently using a MessageBox. However, I thought it would be a good idea to add a Dialog control to abap2UI5 and utilize it in Demo 300. This would provide developers already familiar with UI5 the option to find and use the Dialog control in abap2UI5, making the transition easier for those coming from the UI5 world.

I understand your concerns about keeping the functionality minimal, but I believe extending abap2UI5 with a Dialog control would align better with the UI5 developers' expectations. Developers are more likely to look for something called "dialog" rather than "popup," as it mirrors the UI5 terminology. That way, abap2UI5 would feel more intuitive for them.

I can certainly explore the possibility of using an existing popup or creating a new one for the Dialog, but from a naming perspective, "dialog" would be much clearer for UI5 developers than "popup."

Let me know your thoughts!

Thanks,
Francisco.

**Comment by oblomov-dev** [comment]
Hi @francisco-milan,

thanks for the demo, sample 300 looks nice. But on my side no messagebox is triggered?

You can also check Z2UI5_CL_DEMO_APP_008 there the functionality is implemented with:
<img width="725" alt="Google Chrome 2024-10-14 09 31 05" src="https://github.com/user-attachments/assets/7aad22ff-88a1-4eb2-ade2-b3854caed8de">

it is exactly the same as in the demo kit:
https://sapui5.hana.ondemand.com/sdk/#/entity/sap.m.MessageBox/sample/sap.m.sample.MessageBox/code

Also the naming fits to the ui5 demo kit with one exception: No events are triggered, means the inline message_used by client->message_box_display can only be used for simple displaying messages.

If you need more functionality than in demo 008, i would prefer developing an own dialog_box (popup). You see here a list of the already existing ones:
https://github.com/abap2UI5/abap2UI5/tree/main/src/02/01
you can just add a new class or adjust an existing one.

Dialog/Popup: Yes naming, i completely understand that you prefer to use the word "dialog", it is common for every UI5 developer. Here i used the word popup in every class refering to the former popup function modules (POPUP_TO_CONFIRM etc), because it is more common to abap developers. not sure which way is better but in the end i can not change the names anymore because this would lead to renaming efforts with the next pull for everyone who is already using abap2UI5.

Not sure if i completely got your point, let me know if you need further information.

**Comment by francisco-milan** [comment]
Hi @oblomov-dev,

Thank you for your feedback on demo 300!

I wanted to explain the differences that I observed between demo 300 and the UI5 Demo Kit sample:

![image](https://github.com/user-attachments/assets/43b9a946-f395-450d-9f90-2be5aa0511e5)
1. Title and Icon: In demo 300 (abap2UI5), the title is set to **"Information"** with a blue information icon, while in the UI5 Demo Kit, the title is **"Error description"** without icon. However, in demo 300, even though I explicitly set the title to **"Error description"** this setting doesn’t seem to take effect, and the **"Information"** title is still displayed. 

![image](https://github.com/user-attachments/assets/0c7db231-b919-4c82-8949-cb4b25ac98ce)

2. Button Styling: The "OK" button in demo 300 has its type set to **"Emphasized"**, making it stand out, while in the UI5 Demo Kit example, the "OK" button uses the default type, which does not have any special styling.

Regarding the lack of action when clicking the **"Damaged"** link, that does seem odd. I’ve attached a video where you can see that clicking the link in demo 300 triggers the message box correctly on my side. This might help in understanding why it’s not being triggered for you.

https://github.com/user-attachments/assets/c8d13e50-31a5-4922-8aa2-7f75fbd677ee

I will also review the Z2UI5_CL_DEMO_APP_008 sample you mentioned for more functionality.

Lastly, I understand the use of "popup" in the context of abap2UI5 as it resonates more with traditional ABAP developers, though "dialog" is indeed more familiar to UI5 developers. I’m fine with the naming convention you’ve used, as changing it now would affect the consistency of the project.

Let me know if this clarifies the changes that I would like to apply or if you need further details.

Best regards,
Francisco.

**Comment by choper725** [comment]
hey all
MsgBox is implemented fully in view1_js file,
and should trigger upon not sending a type to the message box,
not sure why it behaves like that in demo 300,
probebly a abap thing, cant really check as im on mobile, but ill start with abap declaration type defaults,
as js code looks ok

**Comment by choper725** [comment]
@francisco-milan  try sending type=‘’ in message_box_display

**Comment by francisco-milan** [comment]
Hi @choper725,

Thanks a lot for the tip! After making the change, the titile is working perfectly. The only difference now is that the "OK" button in my demo has its type set to **"Emphasized"**, whereas in the UI5 Demo Kit sample, the button has the default setting with no special styling.


![image](https://github.com/user-attachments/assets/eb8aa3d2-c593-46b4-9f39-72772286620c)

![image](https://github.com/user-attachments/assets/6cfbdda0-da1d-4ea4-afec-92e22489044d)

I appreciate your help!

Best regards,
Francisco.

**Comment by choper725** [comment]
hi @francisco-milan ,
glad it helped, 
for the emphasized, check if property EMPHASIZEDACTION is available to send ‘OK’ to it,
im not infront of the system to check that..

**Comment by francisco-milan** [comment]
Hi @choper725,

Thanks again for your suggestion! I tried several combinations for both EMPHASIZEDACTION and INITIALFOCUS, including values like "OK", "CANCEL", "NO", " ", "no", "ok", and "cancel", but none of them removed the special styling on the "OK" button—it still remains Emphasized.

![image](https://github.com/user-attachments/assets/089bf0f9-0894-4f46-a809-ca7ec5bb1bd4)

If you have any other ideas, I’d love to hear them!

Best regards,
Francisco.

**Comment by choper725** [comment]
oh i think i know what, i think default value fot emhasize is OK, try sending a ‘Okay’ to actions parameter, i think it will render not emphasized

**Comment by francisco-milan** [comment]
Hi @choper725,

Thanks for the suggestion! I tried your approach, but I ended up applying a little trick to avoid the "OK" button being emphasized—I added a space after "OK." It works, but I'm not sure if this is causing the button in our abap2UI5 demo to appear slightly bigger than the "OK" button in the UI5 Demo Kit sample.

![image](https://github.com/user-attachments/assets/3fe86abb-216a-454f-9b70-9aa449e1d1ac)

![image](https://github.com/user-attachments/assets/40a1711e-63e4-48a9-b105-4fb6ea38a13c)

If there's no quick fix to make the "OK" button smaller, I think it's something we can live with for now.

Also, I’ve added some comments to the code to make it easier for anyone who might need to apply the same solution in the future. Hopefully, it will help in understanding how this demo works.

Thanks again for your help!

Best regards,
Francisco.
